geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaCacheLoader.scala

```diff
@@ -155,28 +155,35 @@ object KafkaCacheLoader {
                               serializer: GeoMessageSerializer,
                               toLoad: KafkaCacheLoaderImpl) extends ThreadedConsumer with Runnable {
 
-    private val loadCache = new ConcurrentHashMap[String, SimpleFeature](1000)
+    private val cache = KafkaFeatureCache.nonIndexing(toLoad.cache)
 
     // track the offsets that we want to read to
     private val offsets = new ConcurrentHashMap[Int, Long]()
+    private var latch: CountDownLatch = _
     private val done = new AtomicBoolean(false)
 
     override protected def closeConsumers: Boolean = false
 
     override protected def consume(record: ConsumerRecord[Array[Byte], Array[Byte]]): Unit = {
       if (done.get) { toLoad.consume(record) } else {
         val message = serializer.deserialize(record.key, record.value)
-        val timestamp = try { record.timestamp() } catch { case e: NoSuchMethodError => System.currentTimeMillis() }
+        val timestamp = try { record.timestamp() } catch { case _: NoSuchMethodError => System.currentTimeMillis() }
         logger.trace(s"Consumed message [$topic:${record.partition}:${record.offset}] $message")
         message match {
-          case m: Change => toLoad.fireEvent(m, timestamp); loadCache.put(m.feature.getID, m.feature)
-          case m: Delete => toLoad.fireEvent(m, timestamp); loadCache.remove(m.id)
-          case m: Clear  => toLoad.fireEvent(m, timestamp); loadCache.clear()
+          case m: Change => toLoad.fireEvent(m, timestamp); cache.put(m.feature)
+          case m: Delete => toLoad.fireEvent(m, timestamp); cache.remove(m.id)
+          case m: Clear  => toLoad.fireEvent(m, timestamp); cache.clear()
           case m => throw new IllegalArgumentException(s"Unknown message: $m")
         }
         // once we've hit the max offset for the partition, remove from the offset map to indicate we're done
-        if (offsets.getOrDefault(record.partition, Long.MaxValue) <= record.offset) {
+        val maxOffset = offsets.getOrDefault(record.partition, Long.MaxValue)
+        if (maxOffset <= record.offset) {
           offsets.remove(record.partition)
+          latch.countDown()
+          logger.info(s"Initial load: consumed [$topic:${record.partition}:${record.offset}] of $maxOffset," +
+              s"${latch.getCount} partitions remaining")
+        } else if (record.offset % 1048576 == 0) { // magic number 2^20
+          logger.info(s"Initial load: consumed [$topic:${record.partition}:${record.offset}] of $maxOffset")
```

Comment:
Hmm.... is the goal here to log every 2^20 messages?

Reply:
this was originally 1M, @jahhulbert-ccri asked it to be changed to a power of 2. the original idea was just to ensure that there was some indicator during the initial load that it was doing something